backend/app/Services/Logistics/PalletMovementService.php

```
@@ -0,0 +1,46 @@
+{
+    public function record(
+        Pallet $pallet,
+        Worker $operator,
+        string $toLocation,
+        ?User $recordedBy = null,
+        ?string $notes = null,
+    ): PalletMovement {
+        return DB::transaction(function () use ($pallet, $operator, $toLocation, $recordedBy, $notes) {
+            // Lock + re-read so the from_location we capture is the real current
+            // location even under concurrent moves of the same pallet.
+            $locked = Pallet::whereKey($pallet->getKey())->lockForUpdate()->firstOrFail();
```

PHPStan (2.2.5)

[error] Call to an undefined static method App\Models\Pallet::whereKey().


(staticMethod.notFound)